packages/fluentui/react-northstar/src/themes/teams/components/Chat/chatMessageStylesComfyRefresh.ts

```diff
@@ -0,0 +1,232 @@
+import { isNil } from 'lodash';
+import { ComponentSlotStylesPrepared, ICSSInJSStyle } from '@fluentui/styles';
+import { getBorderFocusStyles } from '../../getBorderFocusStyles';
+import { chatMessageSlotClassNames, ChatMessageStylesProps } from '../../../../components/Chat/ChatMessage';
+import { pxToRem } from '../../../../utils';
+import { screenReaderContainerStyles } from '../../../../utils/accessibility/Styles/accessibilityStyles';
+import { ChatMessageVariables } from './chatMessageVariables';
+
+const displayActionMenu = (overlayZIndex: ICSSInJSStyle['zIndex']): ICSSInJSStyle => ({
+  zIndex: overlayZIndex!,
+  overflow: 'visible',
+  opacity: 1,
+  width: 'auto',
+});
+
+export const chatMessageStylesComfyRefresh: ComponentSlotStylesPrepared<
+  ChatMessageStylesProps,
+  ChatMessageVariables
+> = {
+  root: ({ props: p, variables: v, theme }): ICSSInJSStyle => {
+    const borderFocusStyles = getBorderFocusStyles({
+      borderRadius: 'inherit',
+      variables: theme.siteVariables,
+    });
+
+    // Fixes the bubble focus border rendering on top of the user avatar
+    delete (borderFocusStyles[':focus-visible'][':before'] as ICSSInJSStyle).zIndex;
+    delete (borderFocusStyles[':focus-visible'][':after'] as ICSSInJSStyle).zIndex;
+
+    return {
+      display: 'flex',
+      flexDirection: 'column',
+      alignItems: p.mine ? 'flex-end' : 'flex-start',
+      outline: 'none',
+
+      [`&:focus-visible .${chatMessageSlotClassNames.timestamp}`]: {
+        opacity: 1,
+      },
+
+      [`&:focus-visible .${chatMessageSlotClassNames.bubble}`]: borderFocusStyles[':focus-visible'],
+    };
+  },
+
+  header: ({ props: p, theme }): ICSSInJSStyle => ({
+    display: 'flex',
+    width: '100%',
+    justifyContent: p.mine ? 'flex-end' : 'start',
+    gap: pxToRem(8),
+    '& > div': {
+      paddingTop: pxToRem(8),
+    },
+    color: theme.siteVariables.colorScheme.default.foreground2,
+  }),
+
+  author: ({ props: p }): ICSSInJSStyle => ({
+    ...((p.mine || p.attached === 'bottom' || p.attached === true) && (screenReaderContainerStyles as ICSSInJSStyle)),
+    fontWeight: 400,
+    marginBottom: pxToRem(2),
+    overflow: 'hidden',
+    whiteSpace: 'nowrap',
+    textOverflow: 'ellipsis',
+  }),
+
+  timestamp: ({ variables: v }): ICSSInJSStyle => {
+    const styles: ICSSInJSStyle = {
+      display: 'inline-block',
+      alignSelf: 'self-start',
+      whiteSpace: 'nowrap',
+      opacity: 0,
+    };
+    if (v.hasReducedHorizontalSpace) {
+      styles.fontSize = '1rem';
+      styles.marginTop = pxToRem(3);
+      styles.marginLeft = pxToRem(2.5);
+      styles.marginRight = pxToRem(2.5);
+      styles.marginBottom = 0;
+    }
```

Comment:
Let's avoid mutations there:

```suggestion
      return {
        ...styles,
        fontSize: '1rem',
        margin: `${pxToRem(3)} ${pxToRem(2.5)} 0 ${pxToRem(2.5)}`
      };
    }
```

Reply:
Fair. I'm going to remove this anyways since it can be implemented in userland and isn't clearly defined here.